spec/src/main/asciidoc/config.asciidoc

```diff
@@ -56,17 +56,34 @@ has a class with the same name.
 The result is that we should have exclusive class names in MicroProfile for classes that might need configuration override in the process environment. 
 The 'first' specification that owns a class name should own that name exclusively. 
 
-Where user class names appear, these are expected to be package qualified.
+Where application class names appear, these are expected to be package qualified.
 
 [[overloadedMethods]]
 ==== Properties That Relate To Overloaded Methods
 
 There are places in the configuration where a method name is used. 
 Accounting for methods with the same name but different signatures
-can be done using indexing. 
-The "1" in the Property name below refers to the occurrence of the 'method' named 
+does not have a recommended approach yet.
+
+[[parameterAnnotations]]
+==== Properties That Relate To Annotated Parameters in Application Methods
+
+Parameter names are not guaranteed to be available during annotation processing.
+Due to this, it can be problematic to rely on parameter names for annotations that
+relate to application methods.
+An annotated application method parameter can be referred to using a positional index, starting at '1'. 
+The "1" in the Property name below refers to the first parameter
 ----
-`org.eclipse.microprofile.example.ExampleBean/setCompletableFuture/1/ManagedExecutorCon fig/maxAsync=5`
+   org.eclipse.microprofile.example.ExampleBean/setCompletableFuture/1/ManagedExecutor/maxAsync=5
```

Comment:
Sorry to be nitpicky here, but I should point out that while ManagedExecutor is an existing class within MicroProfile, it isn't an annotation, and this section is about overriding annotation attributes, so this could cause some confusion.   One idea would be to make this example more generic,
```
org.eclipse.microprofile.example.ExampleBean/setSomething/1/SomeMicroProfileAnno/someAttribute=5
```

Reply:
I love nitpicky. I am replacing my text with yours as it is better.